### PR TITLE
Feat(eos_designs): Add hardware_counters schema

### LINKED ISSUE #2837: Add schema for `hardware_counters`

### PR DESCRIPTION
**Pull request by chetryan**
## Change Summary

Add schema for `hardware_counters` in `eos_designs`.

## Related Issue(s)

Fixes #2837 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Update the `eos_designs` schema

## How to test

- run `molecule test -s eos_designs_unit_tests`

### User Checklist

- Review if all the required changes were made.
- Review whether further documentation updates are required.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests_v4.0/inventory/group_vars/DC1_FABRIC.yml [hunk 1]
```diff
@@ -308,3 +308,10 @@ platform_speed_groups:
 boot:
   secret:
     key: a153de6290ff1409257ade45f
+
+hardware_counters:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests_v4.0/inventory/group_vars/DC1_FABRIC.yml]: Please move this test to `eos_designs_unit_tests` and create a single hostvars file to create a minimal config to only test this.
Look at some of the other hostvars examples in there.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                This validation IS NOT made by the schemas.
+            layer3:
+              type: bool
+              description: |
+                Supported only for the 'ip' feature
+            vrf:
+              type: str
+              description: |
+                Supported only for the 'route' feature.
+                This validation IS NOT made by the schemas.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            prefix:
+              type: str
+              description: |
+                Supported only for the 'route' feature.
+                Mandatory for the 'route' feature.
+                This validation IS NOT made by the schemas.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+              valid_values:
+                - ipv4
+                - ipv6
+                - mac
+              description: |
+                Supported only for the following features:
+                - acl: [ipv4, ipv6, mac] if direction is 'out'
+                - multicast: [ipv4, ipv6]
+                - route: [ipv4, ipv6]
+                This validation IS NOT made by the schemas.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+            direction:
+              type: str
+              valid_values:
+                - in
+                - out
+                - cpu
+              description: |
+                Most features support only 'in' and 'out'. Some like traffic-policy support 'cpu'.
+                Some features DO NOT have any direction.
+                This validation IS NOT made by the schemas.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+                - vtep encap
+            direction:
+              type: str
+              valid_values:
+                - in
+                - out
+                - cpu
+              description: |
+                Most features support only 'in' and 'out'. Some like traffic-policy support 'cpu'.
+                Some features DO NOT have any direction.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]:
```suggestion
                Some features do not have any direction.
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests_v4.0/inventory/group_vars/DC1_FABRIC.yml (hunk 1 above)
Comment (chetryan) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests_v4.0/inventory/group_vars/DC1_FABRIC.yml]: Moved to `eos_designs_unit_tests`

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml (hunk 3 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]: Using $ref to point to the `eos_cli_config_gen` schema.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml (hunk 5 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]: Using $ref to point to the `eos_cli_config_gen` schema.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml (hunk 6 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]: Using $ref to point to the `eos_cli_config_gen` schema.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml (hunk 4 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]: Using $ref to point to the `eos_cli_config_gen` schema.

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml (hunk 2 above)
Comment (chetryan) [reply ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/hardware_counters.schema.yml]: Using $ref to point to the `eos_cli_config_gen` schema.

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Pull request merged**